plugins/ufm_log_analyzer_plugin/src/loganalyze/log_analyzers/events_log_analyzer.py

```diff
@@ -61,17 +46,66 @@ def get_events_by_log_level(self, log_level="CRITICAL"):
 
         return self._log_data_sorted[self._log_data_sorted["severity"] == log_level]
 
-    def get_events_by_log_level_and_event_types_as_count(self, log_level="CRITICAL"):
-        if log_level not in self._supported_log_levels:
-            log.LOGGER.error(
-                f"Requested log level {log_level} is not valid, "
-                f"options are {self._supported_log_levels}"
-            )
-            return None
-        events_by_log_level = self.get_events_by_log_level(log_level)
-        return events_by_log_level["event_type"].value_counts()
+    def plot_top_n_critical_events_over_time(self, n=10):
+        critical_events = self.get_events_by_log_level("CRITICAL")
+        total_critical_events = critical_events.groupby("event").size().reset_index(name="count")
+
+        # Get the top 5 events with the highest count overall
+        top_n_events = total_critical_events.nlargest(n, 'count')
 
-    def print_critical_events_per_hour(self):
+        # Group the top 5 events by time interval
+        critical_events_grouped_by_time = \
+            critical_events[critical_events["event"].isin(top_n_events["event"])]\
+            .groupby([DataConstants.AGGREGATIONTIME, "event"])\
+            .size().reset_index(name="count")
+
+        pivot_top_n_events_by_hour = critical_events_grouped_by_time.pivot(
+            index=DataConstants.AGGREGATIONTIME, columns="event", values="count"
+        ).fillna(0)
```

Comment:
Consider adding a check to handle cases where fewer than `n` critical events exist, which might cause the pivot or plot to fail.

Reply:
Actually I tested it when I run the logic. It was able to plot it even when there is less